src/main/java/ch/wisv/areafiftylan/products/service/OrderServiceImpl.java

```diff
@@ -157,8 +165,12 @@ public Order removeTicketFromOrder(Long orderId, TicketType type, boolean pickup
         }
     }
 
-    private static Predicate<Ticket> isEqualToInput(TicketType type, boolean pickupService, boolean chMember) {
-        return t -> (t.getType() == type) && (t.isChMember() == chMember) && (t.hasPickupService() == pickupService);
+    private static Predicate<Ticket> isEqualToInput(String type, List<String> options) {
+
+        return t -> (t.getType().getName().equals(type) && t.getEnabledOptions().stream().
```

Comment:
If you make it a set, you can check on `containsAll` and if the `size` are equal